sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -102,3 +109,18 @@ object LogicalRelation {
     LogicalRelation(relation, toAttributes(schema), Some(table), false)
   }
 }
+
+/**
+ * Extract the [[BaseRelation]] and [[CatalogTable]] from [[LogicalRelation]]. This also provides
+ * [[LogicalRelation]] itself, to cover the case where other params are needed (but not required
+ * to have specific value to match).
+ */
+object RelationAndCatalogTable {
+  def unapply(plan: LogicalPlan): Option[(LogicalRelation, BaseRelation, Option[CatalogTable])] = {
```

Comment:
```suggestion
  def unapply(plan: LogicalRelation): Option[(BaseRelation, Option[CatalogTable])] = {
```

How about this? Then the caller can be simplified `l @ RelationAndCatalogTable(fsRelation: HadoopFsRelation, _)`

Reply:
Yeah if this work, then looks good, especially with your suggested name. The new name implies they could get the LogicalRelation via `l @ LogicalRelationWithTable(...)`.